pkg/incentives/contract/redistribution.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package contract
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/log"
+	"github.com/ethersphere/bee/pkg/transaction"
+	"github.com/ethersphere/go-sw3-abi/sw3abi"
+)
+
+var (
+	//TODO: add correct ABI of redistribution contract
+	incentivesContractABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+
+	ErrHasNoStake = errors.New("has no stake")
+)
+
+const loggerName = "redistributionContract"
+
+type Interface interface {
+	ReserveSalt(context.Context) ([]byte, error)
+	IsPlaying(context.Context, uint8) (bool, error)
+	IsWinner(context.Context) (bool, error)
+	Claim(context.Context) error
+	Commit(context.Context, []byte) error
+	Reveal(context.Context, uint8, []byte, []byte) error
+}
+
+type Service struct {
+	overlay                   common.Hash
```

Comment:
should `overlay` be `common.Hash` or `swarm.Address`?

Reply:
swarm.Address. Thankyou for highlighting